src/ui/public/validate_query/validate_query.js

```diff
@@ -32,40 +32,37 @@ define(function (require) {
           };
 
           function validator(query) {
-            var index;
-            var type;
             if (request.abort) request.abort();
```

Comment:
Hmm... I'm not sure if this ever worked then.  The promise that was set on `request` wasn't the raw promise returned by the es client even before these changes.

This is what existed before:

![screen shot 2015-10-22 at 1 05 39 pm](https://cloud.githubusercontent.com/assets/53814/10672367/9f96d6da-78bd-11e5-8db7-32b3ad2e7549.png)

So `request` was actually set to a new promise that was returned by `.then(success, error)`.

If that was indeed a bug already, this PR has a bit of urgency to it since it's needed for other PRs, so I'd rather address the bug in a subsequent PR.